Mathlib/AlgebraicGeometry/EllipticCurve/Group.lean

```diff
@@ -642,11 +648,46 @@ end Point
 
 end WeierstrassCurve.Affine
 
-namespace EllipticCurve.Affine
+namespace WeierstrassCurve.Jacobian.Point
 
-/-! ## Elliptic curves in affine coordinates -/
+/-! ## Weierstrass curves in Jacobian coordinates -/
 
-namespace Point
+variable {F : Type u} [Field F] {W : Jacobian F}
+
+lemma zero_add (P : W.Point) : 0 + P = P :=
+  (toAffineAddEquiv W).injective <| by
+    simp only [map_add, toAffineAddEquiv_apply, toAffineLift_zero, _root_.zero_add]
+
+lemma add_zero (P : W.Point) : P + 0 = P :=
+  (toAffineAddEquiv W).injective <| by
+    simp only [map_add, toAffineAddEquiv_apply, toAffineLift_zero, _root_.add_zero]
+
+lemma add_left_neg (P : W.Point) : -P + P = 0 :=
+  (toAffineAddEquiv W).injective <| by
+    rcases P
+    simp only [map_add, toAffineAddEquiv_apply, toAffineLift_neg, _root_.add_left_neg,
+      toAffineLift_zero]
+
+lemma add_comm (P Q : W.Point) : P + Q = Q + P :=
+  (toAffineAddEquiv W).injective <| by simp only [map_add, _root_.add_comm]
+
+lemma add_assoc (P Q R : W.Point) : P + Q + R = P + (Q + R) :=
+  (toAffineAddEquiv W).injective <| by simp only [map_add, _root_.add_assoc]
+
+noncomputable instance instAddCommGroupPoint : AddCommGroup W.Point where
```

Comment:
I think you can just inline all the lemmas and remove the namespace section:
```suggestion
noncomputable instance WeierstrassCurve.Jacobian.instAddCommGroupPoint
    {F : Type u} [Field F] {W : Jacobian F} : AddCommGroup W.Point where
```
I don't think anyone would use those lemmas, and they confuse Lean if not protected.

Reply:
Good point - I'll hide the affine ones too. I'll keep the namespace because I'll delete my instance name.